clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -339,6 +339,7 @@ private void closeHeartbeatThread() {
         try {
             thread.join();
         } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
I think we do not need to reset here since we are throwing the `InterruptException` again, which will set the flag as well.

Reply:
Right.